core/src/main/java/ar/com/wolox/wolmo/core/permission/PermissionManager.java

```diff
@@ -0,0 +1,89 @@
+package ar.com.wolox.wolmo.core.permission;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.os.Handler;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.app.Fragment;
+import android.support.v4.content.ContextCompat;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class PermissionManager {
+
+    private static PermissionManager sInstance;
+
+    private int sRequestCount = 1;
+    private HashMap<Integer, PermissionListener> sRequestListeners = new HashMap<>();
+
+    private PermissionManager() {};
+
+    public synchronized static PermissionManager getInstance() {
+        if(sInstance == null) {
+            sInstance = new PermissionManager();
+        }
+        return sInstance;
+    }
+
+    public boolean requirePermission(Fragment fragment, PermissionListener listener, String... permissions) {
+        String[] ungrantedPermissions = filterUngranted(fragment.getActivity(), permissions);
+        if (ungrantedPermissions.length > 0) {
+            fragment.requestPermissions(ungrantedPermissions, sRequestCount);
+            sRequestListeners.put(sRequestCount++, listener);
+            return false;
+        }
+        listener.onPermissionsGranted();
+        return true;
+    }
+
+    public boolean requirePermission(Activity activity, PermissionListener listener, String... permissions) {
+        String[] ungrantedPermissions = filterUngranted(activity, permissions);
+        if (ungrantedPermissions.length > 0) {
+            ActivityCompat.requestPermissions(activity, ungrantedPermissions, sRequestCount);
+            sRequestListeners.put(sRequestCount++, listener);
```

Comment:
Why is there a `sRequestCount++` here? 
In the line above, there is a request using `sRequestCount` (without the ++), which would mean that the request code sent to the `Activity` won't match the one associated with the listener in the `sRequestListeners` map.

This is important because on the `onRequestPermissionsResult` method the listener is being removed from the map based on the request code.

Am I missing something?